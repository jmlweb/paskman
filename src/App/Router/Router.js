// @flow
import React from 'react';
import { Route } from 'react-router-dom';
import { type RoutesType } from '../../types';

type Props = {
  routes: RoutesType,
};

const Router = (props: Props) => (
  <main>
    {props.routes.map(
      route => <Route key={`route${route.title}`} exact path={route.path} component={route.component} />
    )}
  </main>
);

export default Router;
