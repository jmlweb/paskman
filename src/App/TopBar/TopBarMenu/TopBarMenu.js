import React from 'react';
import PT from 'prop-types';
import { Link } from 'react-router-dom';
import styled, { css } from 'styled-components';
import styledMap from 'styled-map';
import colors from '../../../styles/colors';
import timings from '../../../styles/timings';
import topBarHeight from '../../../styles/topBarHeight';
import { RoutesPT } from '../../../propTypes';

const StyledLink = styled(Link)`
  align-items: center;
  border-bottom: 1px solid ${colors.secondaryS2};
  box-shadow: 0 1px 0 ${colors.secondaryT2};
  color: ${colors.white};
  display: flex;
  filter: blur(5px);
  flex: 1;
  flex-direction: column;
  font-size: 2rem;
  justify-content: center;
  text-decoration: none;
  transition: all 0.4s ${timings.easeInOutQuad};
  &:active {
    background: ${colors.secondaryS7};
  }
  &:last-child {
    border-bottom: 2px solid transparent;
  }
`;

const openMenu = css`
  border-bottom: 1px solid ${colors.secondaryS5};
  border-top: 1px solid ${colors.secondaryS5};
  max-height: calc(100vh - ${topBarHeight});
  transition: max-height 0.3s ${timings.easeOutBack};
`;

const closedMenu = css`
  max-height: 0;
  transition: max-height 0.7s ${timings.easeOutBack};
`;

const StyledMenu = styled.nav`
  background: ${colors.secondary};
  bottom: 0;
  box-shadow: 0 7px 6px ${colors.blackA6};
  color: ${colors.white};
  display: flex;
  flex-direction: column;
  left: 0;
  margin: 0;
  overflow: hidden;
  padding: 0;
  position: fixed;
  right: 0;
  top: ${topBarHeight};
  z-index: 5;
  ${styledMap({
    menuOpen: openMenu,
    default: closedMenu,
  })};
  ${StyledLink} {
    filter: blur(0);
    opacity: ${styledMap({
    menuOpen: 1,
    default: 0,
  })};
  }
`;

const getLink = toggleMenu => route =>
  <StyledLink key={`link${route.title}`} to={route.path} onClick={toggleMenu}>{route.title}</StyledLink>;

const Menu = ({ menuOpen, toggleMenu, routes }) => (
  <StyledMenu menuOpen={menuOpen}>
    {routes.map(getLink(toggleMenu))}
  </StyledMenu>
);

Menu.defaultProps = {
  menuOpen: false,
  routes: [],
};

Menu.propTypes = {
  menuOpen: PT.bool,
  toggleMenu: PT.func.isRequired,
  routes: RoutesPT,
};

export default Menu;
