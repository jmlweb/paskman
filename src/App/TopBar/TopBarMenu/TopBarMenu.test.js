import React from 'react';
import {
  BrowserRouter,
} from 'react-router-dom';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import TopBarMenu from './TopBarMenu';
import Router from '../../Router/Router';
import createDynamicComponent from '../../../components/DynamicComponent/DynamicComponent';

const routes = [
  {
    path: '/',
    title: 'Home',
    component: createDynamicComponent(
      () => import('../../../scenes/Dashboard/Dashboard'),
      () => require('../../../scenes/Dashboard/Dashboard'),
    ),
  },
];

describe('Menu', () => {
  let wrapper;
  let count = 0;
  beforeEach(() => {
    wrapper = mount(
      <BrowserRouter key={`b${count}`}>
        <div key={`d${count}`}>
          <TopBarMenu key={`m${count}`} routes={routes} />
          <Router key={`r${count}`} routes={routes} />
        </div>
      </BrowserRouter>
    );
    count += 1;
  });
  it('Render the component', () => {
    expect(wrapper.find('Menu').length).toEqual(1);
  });
  it('Matchs snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
