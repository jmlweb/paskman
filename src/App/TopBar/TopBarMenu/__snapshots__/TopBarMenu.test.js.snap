// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Matchs snapshot 1`] = `
<div>
  <nav
    className="sc-bwzfXH gjZsxj"
  >
    <a
      className="sc-bdVaJa ioMEjV"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </nav>
  <main />
</div>
`;
